.github/workflows/publish-staging-redacted-export.yml

```
@@ -0,0 +1,160 @@
+          CI: true
+        run: |
+          bin/rails redacted_export:generate_sql > tmp/create_redacted_schema.sql
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: create_redacted_schema.sql
+          path: tmp/create_redacted_schema.sql
+          retention-days: 7
+
+  export_database:
+    name: Export and upload redacted database dump
+    runs-on: ubuntu-latest
+    needs: create_redacted_schema_sql
+
+    steps:
+      - name: Install cf client
+        env:
+          CF_CLI_VERSION: v7
+        run: |
+          mkdir -p $GITHUB_WORKSPACE/bin
+          curl -L "https://packages.cloudfoundry.org/stable?release=linux64-binary&version=${CF_CLI_VERSION}" | tar -zx -C $GITHUB_WORKSPACE/bin
+          echo "$GITHUB_WORKSPACE/bin" >> $GITHUB_PATH
+
+      - name: Install conduit plugin
+        run: cf install-plugin conduit -f
+
+      - name: Install PostgreSQL client
+        run: sudo apt-get install libpq-dev
+
+      - uses: actions/download-artifact@v2
+        with:
+          name: create_redacted_schema.sql
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.DB_UPLOAD_STAGING_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.DB_UPLOAD_STAGING_AWS_SECRET_ACCESS_KEY }}
+          aws-region: eu-west-2
+
+      - name: Export database
+        id: export-database
+        env:
+          SPACE: staging
+          CF_USERNAME: ${{ secrets.PaaSUsernameStaging }}
+          CF_PASSWORD: ${{ secrets.PaaSPasswordStaging }}
+          S3_BUCKET: psd-staging-database-dumps
+        run: |
+          TIMESTAMP=$(date +'%F-%H-%M-%S')
+          TEMP_DB_SERVICE_NAME=psd-redacted-export-$TIMESTAMP
+          FINAL_S3_URL="s3://${S3_BUCKET}/${TIMESTAMP}.sql.gz"
+          echo "::set-output name=temp-db-service-name::$TEMP_DB_SERVICE_NAME"
+          echo "::set-output name=final-s3-url::$FINAL_S3_URL"
+          cf api api.london.cloud.service.gov.uk
+          cf auth
+          cf target -o 'beis-opss' -s $SPACE
+          PSD_DATABASE_GUID=$(cf service psd-database --guid)
+          PSD_DATABASE_PLAN=$(cf service psd-database | awk -F ':' '$1=="plan" { gsub(/ /,"", $NF); print $NF }')
+          cf create-service postgres $PSD_DATABASE_PLAN $TEMP_DB_SERVICE_NAME -c "{\"restore_from_latest_snapshot_of\": \"$PSD_DATABASE_GUID\"}"
+          echo "::group::Waiting for db"
+          until cf service $TEMP_DB_SERVICE_NAME > /tmp/db_exists && grep -E "create succeeded|update succeeded" /tmp/db_exists; do sleep 30; echo "Waiting for db..."; done
+          echo "::endgroup::"
+          echo "::group::Schema dump"
+          cf conduit $TEMP_DB_SERVICE_NAME -- psql < create_redacted_schema.sql
+          cf conduit $TEMP_DB_SERVICE_NAME -- pg_dump --schema=redacted --file=psd_redacted_export.sql --no-acl --no-owner --quote-all-identifiers --format=p --inserts --encoding=UTF8
```

Suggestion: I wonder if it would be more efficient to combine these into one command i.e.

```suggestion
          cf conduit $TEMP_DB_SERVICE_NAME -- psql < create_redacted_schema.sql && pg_dump --schema=redacted --file=psd_redacted_export.sql --no-acl --no-owner --quote-all-identifiers --format=p --inserts --encoding=UTF8
```

---

Resolving as the PR no longer uses conduit, but very likely, thanks for the tip!